java/vector/src/test/java/org/apache/arrow/vector/TestValueVector.java

```diff
@@ -315,8 +317,22 @@ public void testBitVector() {
       assertEquals(0, accessor.get(0));
       assertEquals(1, accessor.get(1));
 
+      // should not change
+      assertEquals(1022, accessor.getNullCount());
+
       // Ensure unallocated space returns 0
       assertEquals(0, accessor.get(3));
+
+      m.set(1, 0);
+      m.set(1022, 0);
+
+      assertEquals(1024, accessor.getNullCount());
+
+      for (int i = 0; i < 1024; ++i) {
+        m.set(i, 1);
+      }
+
+      assertEquals(0, accessor.getNullCount());
```

Comment:
This test could be more rigorous in case there are issues with the bitmasking logic -- perhaps check the null count either for each iteration of the for loop, or periodically if that is too slow -- I only see 2 bits set to 0, or all of them.